cocos/2d/components/ui-opacity.ts

```diff
@@ -106,50 +127,93 @@ export class UIOpacity extends Component {
 
         const render = node._uiProps.uiComp as UIRenderer;
         const uiOp = node.getComponent<UIOpacity>(UIOpacity);
-        let interruptOpacity = interruptParentOpacity;// if there is no UIOpacity component, it should always equal to 1.
 
+        if (recursiveWithoutUiOp && uiOp) {
```

Comment:
`recursiveWithoutUiOp ` means recursing only nodes without `uiopacity` component,  and `uiOp` means it has a `uiopacity` component. It seems conflict here.

Reply:
'uiOp ! = null' is indicating that the UiOpacity component exists. ‘recursiveWithoutUiOp‘ indicates that it doesn't need to exist, so both conditions are met when handling the situation.